### PR TITLE
#9 - Set up Jest and Test Library

### LINKED ISSUE #9: Set up Jest tests in prototype

### PR DESCRIPTION
**Pull request by abbyoung**
## Description 

This PR adds Jest, React Test Library, and their necessary config/dependencies.

Fixes #9 

## Review Notes
To run locally:
```sh
yarn install
yarn test
```

Testing Library is new to me, so please let me know if there's a better way to test a simple image component.

I thought I could test the size to see if it rendered properly, but `next/image` handles sizing in a strange way and there's no attribute to compare my test value against.

### CODE REVIEW

File: jest.config.js [hunk 1]
```diff
@@ -0,0 +1,10 @@
+module.exports = {
+  moduleFileExtensions: ['ts', 'tsx', 'js', 'jsx'],
+  transform: {
+    '^.+\\.(js|jsx|ts|tsx)$': 'babel-jest',
+  },
+  moduleNameMapper: {
+    '\\.(scss|sass|css|png)$': 'identity-obj-proxy',
```
Comment (abbyoung) [inline jest.config.js:7]: This is required if we want to use css modules and/or imported images in testing.

File: jest.config.js [hunk 2]
```diff
@@ -0,0 +1,10 @@
...
+  },
+  setupFilesAfterEnv: ['<rootDir>/jest.setup.js'],
```
Comment (abbyoung) [inline jest.config.js:9]: Without this and the one-liner setup file, component rendering failed.

File: .storybook/preview.js [hunk 3]
```diff
@@ -1,5 +1,14 @@
 import '@trussworks/react-uswds/lib/uswds.css'
 import '@trussworks/react-uswds/lib/index.css'
+import * as NextImage from 'next/image'
+
+// Storybook and next/image component do not play nice together
+// This enables us to use the <Image/> component and still view in Storybook
+const OriginalNextImage = NextImage.default
+Object.defineProperty(NextImage, 'default', {
+  configurable: true,
+  value: (props) => <OriginalNextImage {...props} unoptimized />,
```
Comment (abbyoung) [inline .storybook/preview.js:10]: We obviously don't want `unoptimized` on our web app images, but this is only for Storybook compatibility. Do you see any potential issues with this approach @suzubara ?

Comment (suzubara) [reply .storybook/preview.js:10]: _interesting!!!_ I just went down a little rabbit hole about this. but this fix looks 👍  to me!

File: package.json [hunk 4]
```diff
@@ -9,7 +9,8 @@
     "storybook": "start-storybook -p 6006 -s ./public",
     "storybook:build": "build-storybook",
     "format": "prettier --write .",
-    "lint": "tsc --noEmit && eslint ."
+    "lint": "tsc --noEmit && eslint .",
+    "test": "jest"
```
Comment (suzubara) [inline package.json:13]: Can we add a watch script `"test:watch": "jest --watch"` also?

File: components/Logo/Logo.test.tsx [hunk 5]
```diff
@@ -0,0 +1,15 @@
+
+/**
+ * @jest-environment jsdom
```
Comment (suzubara) [inline components/Logo/Logo.test.tsx:3]: well this prompted me to discover that Jest changed their default env from jsdom to node TIL

File: package.json (hunk 4 above)
Comment (abbyoung) [reply package.json:13]: YES!

**Review by suzubara — APPROVED** [review]
👍  confirmed tests run & Storybook works! It looks like some conflicts need resolving and if you don't mind just making sure the precommit hooks run since those have been merged, this is good to go!

**Review by suzubara — APPROVED** [review]

**Pull request merged**